app/common/data/interfaces/user.py

```diff
@@ -13,6 +15,11 @@ def get_user(id_: str | uuid.UUID) -> User | None:
     return db.session.get(User, id_)
 
 
+def get_current_user() -> User:
+    user = cast(User, current_user)
+    return user
```

Comment:
I got stuck on solving this problem for ages by trying to force Flask's actual `current_user` to have the User type, and never made any progress, which has led to us having to cast current_user to User in a load of places in the codebase.

Just doing this feels like a much more sensible and pragmatic solution to that problem - good idea 🙇 👍

It might be nice to do this as a separate PR and update everywhere in the codebase appropriately? It feels very much an isolated change/refactor that we could do by itself.